### PR TITLE
consider smooth talker in dialogs

### PR DESCRIPTION
**Pull request by dzmitry-rudnouski**

### CODE REVIEW

File: scripts_src/redding/rcascort.ssl [hunk 1]
```diff
@@ -211,7 +211,7 @@ procedure talk_p_proc begin
 
    GetReaction;
 
-   if (dude_iq < 4) then begin
+   if (dude_iq_talk < 4) then begin
```
Comment (burner1024) [inline scripts_src/redding/rcascort.ssl:214]: I don't think it is correct. It's NPC reacting to PC, not PC response.

Comment (dzmitry-rudnouski) [reply scripts_src/redding/rcascort.ssl:214]: Hmm, let me think about it.

Comment (dzmitry-rudnouski) [reply scripts_src/redding/rcascort.ssl:214]: @burner1024 I see some general contradiction between perk description and scripts.
Description: "... Each level of this Perk will increase your intelligence by one, up to 10, for purposes of dialog only."
But there are no example in scripts where dude_smooth_talker var is used as integer (except your commit 91a162ee6e68173ad6e8c9632c21d682c326101e).
In all examples that I see it's used as boolean, e.g. https://github.com/BGforgeNet/Fallout2_Unofficial_Patch/blob/master/scripts_src/newreno/ncjimmyj.ssl#L494
Did I miss anything?

Comment (burner1024) [reply scripts_src/redding/rcascort.ssl:214]: No, that's right, the perk is unimplemented. The existing usage in scripts is just some erratic stuff. It has nothing to do with how perk is supposed to work.

Comment (burner1024) [reply scripts_src/redding/rcascort.ssl:214]: Actually, I was mistaken.
The perk works on engine level, it adds INT in `giQ_Option` function, which is used by `NOption` and similar macros.

**Pull request closed without merging**